string task_name
string[] entity_names
string[] entity_values